mirrulations-extractor/src/mirrextractor/extractor.py

```diff
@@ -0,0 +1,97 @@
+from datetime import datetime
+import os
+import time
+import io
+import pdfminer
+import pdfminer.high_level
+import pikepdf
+
+
+class Extractor:
+    """
+    Class containing methods to extract text from files.
+    """
+    @staticmethod
+    def extract_text(attachment_path, save_path):
+        """
+        This method takes a complete path to an attachment and determines
+        which type of extraction will take place.
+        *Note* save_path is for later use when saving the extracted text
+        Parameters
+        ----------
+        attachment_path : str
+            the complete file path for the attachment that is being extracted
+            ex. /path/to/pdf/attachment_1.pdf
+        save_path : str
+            the complete path to store the extract text
+            ex. /path/to/text/attachment_1.txt
+        """
+        # gets the type of the attachment file
+        #   (ex. /path/to/pdf/attachment_1.pdf -> pdf)
+        file_type = attachment_path[attachment_path.rfind('.')+1:]
+        if file_type.endswith('pdf'):
+            print(f"Extracting text from {attachment_path}")
+            Extractor._extract_pdf(attachment_path, save_path)
+        else:
+            print(f"FAILURE: attachment doesn't have appropriate extension \
+            {attachment_path}")
+
+    @staticmethod
+    def make_save_path(path):
```

Comment:
Could this method be pushed out into the path_generator class rather than using .replace?

Reply:
Method was pushed into the *path_generator* instead of the *extractor* with the same functionality. Since json is not provided in that space, .replace is necessary for editing the provided path to suit the *save_path*.